packages/ruby/lib/readme/webhook.rb

```diff
@@ -0,0 +1,24 @@
+require 'readme/metrics'
+
+module Readme
+  class Webhook
+    def self.verify(body, signature, secret)
+      raise 'Missing Signature' unless signature
+
+      parsed = signature.split(',').each_with_object({ time: -1, readme_signature: '' }) do |item, accum|
+        k, v = item.split('=')
+        accum[:time] = v if k.eql? 't'
+        accum[:readme_signature] = v if k.eql? 'v0'
+      end
+
+      # Make sure timestamp is recent to prevent replay attacks
+      thirty_minutes = 30 * 60
+      raise 'Expired Signature' if Time.now.utc - Time.at(0, parsed[:time].to_i, :millisecond).utc > thirty_minutes
```

Comment:
I *think* you can do the following in rails:

```suggestion
      raise 'Expired Signature' if Time.at(0, parsed[:time].to_i, :millisecond).utc < 30.minutes.ago
```

Reply:
So I did actually have that over here when I wrote it in the rails app: https://github.com/readmeio/metrics-sdks/pull/568/commits/28c76240e759f71a42cda36b2cf706ead8b101ff#diff-548d6c1602d07c9a685cb30d93f75730559117156e630c54971658e7560a6d59R16-R18 

But since that [comes from Rails](https://api.rubyonrails.org/v7.0.3.1/classes/Numeric.html#method-i-minutes), and our module doesnt have a dependency on rails, I decided to refactor it to this. Definitely not as readable, but I didnt fancy bringing in the active_support gem just for this.

Reply:
Gotchya gotchya